dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java

```diff
@@ -142,4 +152,76 @@ public PublishStatus publish ( PublisherConfig config, PublishStatus status ) th
         return status;
     }
 
+    @Override
+    public void addFilterDescriptor(final FilterDescriptor filterDescriptor) {
+        this.loadedFilters.put(filterDescriptor.getKey(),filterDescriptor);
+    }
+
+    @Override
+    public List<FilterDescriptor> getFiltersDescriptorsByRole(final User user) throws DotDataException {
+        if(APILocator.getUserAPI().isCMSAdmin(user)){
+            return new ArrayList<>(this.loadedFilters.values());
+        }
+        final List<Role> roles = APILocator.getRoleAPI().loadRolesForUser(user.getUserId(), true);
+        Logger.info(this,"User Roles: " + roles.toString());
+        final List<FilterDescriptor> filters = new ArrayList<>();
+        for(final Map.Entry<String,FilterDescriptor> filterDescriptorMap : this.loadedFilters.entrySet()){
+            final String filterRoles = filterDescriptorMap.getValue().getRoles();
+            Logger.info(PublisherAPI.class,"File: " +filterDescriptorMap.getKey() + " Roles: " + filterRoles );
+            for(final Role role : roles){
+                if(UtilMethods.isSet(role.getRoleKey()) && filterRoles.contains(role.getRoleKey())){
+                    filters.add(filterDescriptorMap.getValue());
+                }
+            }
+        }
+
+        return filters;
+    }
+
+    @Override
+    public Map<String, FilterDescriptor> getFilterDescriptorMap() {
+        return this.loadedFilters;
+    }
+
+    @Override
+    public FilterDescriptor getFilterDescriptorByKey(final String filterKey) {
+        final FilterDescriptor defaultFilter = this.loadedFilters.values().stream().filter(filterDescriptor -> filterDescriptor.isDefaultFilter()).findAny().get();
+        if(!UtilMethods.isSet(filterKey)){
+            return defaultFilter;
+        }
+        return this.loadedFilters.getOrDefault(filterKey,defaultFilter);
+    }
```

Comment:
This could be refactored by
```
return !UtilMethods.isSet(filterKey)?
     this.loadedFilters.values().stream().filter(FilterDescriptor::isDefaultFilter()).findFirst().get():
    this.loadedFilters.containsKey(filterKey)? this.loadedFilters.get(filterKey) :  this.loadedFilters.values().stream().filter(FilterDescriptor::isDefaultFilter()).findFirst().get();
        
```

In addition you can create a method called getDefaultFilter, that basically returns this
```
 this.loadedFilters.values().stream().filter(FilterDescriptor::isDefaultFilter()).findFirst().get()
```

And call such as this:
```
return !UtilMethods.isSet(filterKey)?
     this.getDefaultFilter():
     this.loadedFilters.containsKey(filterKey)?
         this.loadedFilters.get(filterKey) :  this.getDefaultFilter();
```